vue-app/src/components/TimeLeft.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <span :class="valueClass || 'value'">
+    {{ values[0] }}
+    <span :class="unitClass || 'unit'">
+      {{ units[0] }}
+    </span>
+    <span v-if="units[1].length > 0">
+      {{ units[1].length > 0 && values[1] }}
+      <span :class="unitClass || 'unit'">
+        {{ units[1] }}
+      </span>
+    </span>
+  </span>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { DateTime } from 'luxon'
+import { Prop } from 'vue-property-decorator'
+
+import { getTimeLeft } from '@/utils/dates'
+import { TimeLeft } from '@/api/round'
+
+@Component
+export default class extends Vue {
+  @Prop() date!: DateTime
+  @Prop() valueClass!: string
+  @Prop() unitClass!: string
+
+  timeLeft: TimeLeft = getTimeLeft(this.date)
+  interval
+
+  mounted() {
+    this.interval = setInterval(() => {
+      this.timeLeft = getTimeLeft(this.date)
+    }, 1000)
+  }
+
+  beforeDestroy() {
+    clearInterval(this.interval)
+  }
+
+  get values(): number[] {
+    if (this.timeLeft.days > 0) return [this.timeLeft.days, this.timeLeft.hours]
+    if (this.timeLeft.hours > 0)
+      return [this.timeLeft.hours, this.timeLeft.minutes]
+    if (this.timeLeft.minutes > 0) return [this.timeLeft.minutes, 0]
+    if (this.timeLeft.seconds > 0) return [this.timeLeft.seconds, 0]
+    return [0, 0]
+  }
+
+  get units(): string[] {
+    if (this.timeLeft.days > 0)
+      return [this.unitPlurality('days'), this.unitPlurality('hours')]
+    if (this.timeLeft.hours > 0)
+      return [this.unitPlurality('hours'), this.unitPlurality('minutes')]
+    if (this.timeLeft.minutes > 0) return [this.unitPlurality('minutes'), '']
```

Comment:
What do you think about having seconds in here? 
<img width="342" alt="Screen Shot 2021-08-11 at 9 13 41 PM" src="https://user-images.githubusercontent.com/15589226/129132904-6f3d15c0-68d0-4e2e-b6a5-61e7e8b0a41f.png">
Might be nice to show the realtime whats left for the user. Only reason why I am thinking this is we get to 1 minute (when theres less that 2 minutes, but greater than 1), and then we get to 59 seconds. So seems a little wonky from that perspective.

Reply:
This was spec'd out by Ryan earlier in the project. As I was using it, I also felt the second countdown could be utilized a bit earlier. 

We could easily add a two lines in to switch to showing second under a certain 'minute' threshold if we want... 
so at 59 minutes it just shows minutes... but perhaps under 10 or 5 minutes we start the seconds countdown.

Reply:
I think thats a fair compromise between the two. But, if its not in the design then thats also fair. I was just experimenting a little bit when I noticed that and thought it might be a nice touch.